spec/STS.assumeRole.cannotAssumeWithoutExternalId.spec.ts

```diff
@@ -0,0 +1,29 @@
+import { awsMockCallback, awsMockFailureCallback } from './support'
+import AWSMock = require('aws-sdk-mock')
+import { IAM } from '../src/IAM'
+// import { STS } from 'aws-sdk'
+
+describe("STS#Assume role fails without externalId",() => {
+    afterEach(() => {
+        AWSMock.restore('STS');
+    })
+    it("should fail to assume a role when the externalId is missing",async () => {
+        try {
+            AWSMock.mock('STS','assumeRole',awsMockFailureCallback('test-data/assumeRole-denied.json'))
```

Comment:
Are these only testing the mocks?

Not sure I know a better way of testing, just not sure if these tests are adding anything.  

Documentation is useful, but will they catch an error?

Reply:
Yes they are. It doesn't really differentiate between between an externalId being required and simply not being allowed to assume a role. I will remove it